extensions/ql-vscode/src/cli.ts

```diff
@@ -629,6 +640,9 @@ export class CodeQLCliServer implements Disposable {
     }
   }
 
+  async generateResultsCsv(metadata: { kind: string; id: string; scored?: string }, resultsPath: string, csvPath: string, sourceInfo?: SourceInfo): Promise<void> {
```

Comment:
Hmmm...the `scored` parameter is unused here. I'm not sure if it is even used when generating csv. @henrymercer is it worth passing in the `scored` value as we're doing for sarif interpretation?

Reply:
We're still producing interpreted alerts here AIUI, so yes I think we'd need to forward the `scored` query metadata here so that queries with alert scores are interpreted correctly.

Reply:
I think this is being forwarded correctly? It's one of the keys in the `metadata` parameter, which is passed on and used in line 605. Do you have an example query with the scored metadata I could use to test this? As far as I can tell none of the standard library queries use it unless I'm missing something...

Reply:
Ah...yes. My mistake. It _is_ being used correctly. The scored parameter is something that @henrymercer has been working on to implement ranked queries that order results by confidence. It's still experimental, and it is only enabled when you enter `canary` mode, which explicitly opts into experimental features.

What would be useful here is if you could replace `{ kind: string; id: string; scored?: string }` with `QueryMetadata` from `interface-types.ts` since it really is the same type.

Once that is changed, I think this is good to go in.